test/parallel/test-beforeexit-event.js

```diff
@@ -1,42 +1,41 @@
 'use strict';
-require('../common');
-var assert = require('assert');
-var net = require('net');
-var revivals = 0;
-var deaths = 0;
+const common = require('../common');
+const net = require('net');
 
-process.on('beforeExit', function() { deaths++; });
+const beforeExit = common.mustCall(() => {
+  return;
+}, 4);
+
+const beforeListener = common.mustCall(() => {
+  return;
+}, 3);
+
+process.on('beforeExit', beforeExit);
 
 process.once('beforeExit', tryImmediate);
 
 function tryImmediate() {
-  console.log('set immediate');
-  setImmediate(function() {
-    revivals++;
+  setImmediate(() => {
+    beforeListener();
```

Comment:
What I had in mind instead of this is to wrap the enclosing function in `common.mustCall()`, for example:

```js
setImmediate(common.mustCall(() => {
  process.once('beforeExit', common.mustCall(tryTimer));
}));
```